pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -197,24 +234,25 @@ func TestDesiredRouterDeployment(t *testing.T) {
 			},
 		},
 	}
-
 	proxyNeeded, err := IsProxyProtocolNeeded(ci, infraConfig.Status.PlatformStatus)
 	if err != nil {
 		t.Errorf("failed to determine infrastructure platform status for ingresscontroller %s/%s: %v", ci.Namespace, ci.Name, err)
 	}
+
+	return ci, ingressConfig, infraConfig, apiConfig, networkConfig, proxyNeeded
+}
+
+func TestDesiredRouterDeploymentDefaults(t *testing.T) {
```

Comment:
That's a bit of a misnomer; maybe this test could be called `TestDesiredRouterDeploymentPrivateShardedCustomTLS` or just `TestDesiredRouterDeployment` (using affixes only for the variant tests).